core/src/androidMain/kotlin/gatt/Callback.kt

```diff
@@ -286,8 +284,10 @@ private val Int.connectionStateString: String
         else -> "Unknown($this)"
     }
 
-private fun <E> SendChannel<E>.trySendOrLog(element: E) {
-    trySend(element).getOrElse { cause ->
-        Log.w(TAG, "Callback was unable to deliver $element", cause)
+private fun <E> Logger.trySendOrLog(channel: SendChannel<E>, element: E) {
+    channel.trySend(element).getOrElse { cause ->
+        warn(cause) {
+            message = "Callback was unable to deliver $element"
+        }
```

Comment:
I personally would rather this remain an extension function on `SendChannel` and be moved as a class function (not top-level) so that call sites would look like:

```kotlin
onResponse.trySendOrLog(event)
```